tools/engine_tool/test/test_command_test.dart

```diff
@@ -83,7 +85,7 @@ void main() {
         '//third_party/protobuf:protoc',
       ]);
       expect(result, equals(1));
-      expect(testEnvironment.processHistory.length, lessThan(3));
+      expect(testEnvironment.processHistory.length, lessThan(6));
```

Comment:
Here and in the test above it: This expectation is impacted by implementation details of the test command. I don't think the number of processes executed is the right test, and instead we should just test that test processes are executed and that non-test processes are executed.

I can send out a followup that fixes that. (I realise I also just added this test in a previous patch so I'm 100% to blame for this one :P)

Reply:
Yes, I think this count is too brittle. Better to have something like:

expect(testEnvironment.processHistory, executed('gn desc'));